clippy_lints/src/transmute.rs

```diff
@@ -88,14 +88,32 @@ declare_lint! {
 /// ```rust
 /// let _: char = std::mem::transmute(x); // where x: u32
 /// // should be:
-/// let _: Option<char> = std::char::from_u32(x);
+/// let _ = std::char::from_u32(x).unwrap();
 /// ```
 declare_lint! {
     pub TRANSMUTE_INT_TO_CHAR,
     Warn,
     "transmutes from an integer to a `char`"
 }
 
+/// **What it does:** Checks for transmutes from a `&[u8] to a `&str`.
+///
+/// **Why is this bad?** Not every byte slice is a valid UTF-8 string.
+///
+/// **Known problems:** None.
```

Comment:
- Speed is a known problem, as `from_utf8` needs to check the slice, and its runtime grows linearly with the slice length. This is also the right point in the documentation to mention `from_utf8_unchecked` as alternative that is as fast as the transmute but has a semantically meaningful name.
- The `unwrap` in the suggestion is also some kind of a problem (assuming the byte slice is user input you should most likely return a `Result::Err`), even though I'd say it's a "good" problem: The program will panic where it before had undefined behavior. This is also something I'd mention in the `TRANSMUTE_INT_TO_CHAR` docs.